pkg/obs/services/objects.go

```diff
@@ -53,41 +53,46 @@ func DeleteObjects(c *config.CloudCredentials, bucketName string) error {
 		if err != nil {
 			return err
 		}
-		if len(listOutput.Contents) == 0 {
+		expectedDelLength := len(listOutput.Contents)
+		if expectedDelLength == 0 {
 			break
 		}
 		input := &obs.DeleteObjectsInput{Bucket: bucketName}
-		input.Objects = make([]obs.ObjectToDelete, 0, len(listOutput.Contents))
+		input.Objects = make([]obs.ObjectToDelete, 0, expectedDelLength)
 		for _, object := range listOutput.Contents {
 			input.Objects = append(input.Objects, obs.ObjectToDelete{Key: object.Key})
 		}
-		_, err = client.DeleteObjects(input)
-		if err == nil {
-			continue
+		output, err := client.DeleteObjects(input)
+		if err != nil {
+			return status.Errorf(codes.Internal, "error deleting obs instance %s object: %v", bucketName, err)
 		}
-		if _, ok := err.(obs.ObsError); ok {
-			return status.Errorf(codes.Internal, "Error deleting OBS instance %s object: %v", bucketName, err)
+		if len(output.Deleteds) != expectedDelLength {
+			return status.Errorf(codes.Internal, "Error deleting OBS instance %s object, fail num: %d", bucketName, len(output.Errors))
 		}
 	}
 	return nil
 }
 
-func DeleteObject(c *config.CloudCredentials, bucketName, objectName string) error {
+func DeleteObject(c *config.CloudCredentials, bucketName string, objectsName ...string) (*obs.DeleteObjectsOutput, error) {
 	client, err := getObsClient(c)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	input := &obs.DeleteObjectsInput{Bucket: bucketName}
 	input.Objects = make([]obs.ObjectToDelete, 0)
-	input.Objects = append(input.Objects, obs.ObjectToDelete{Key: objectName})
-	_, err = client.DeleteObjects(input)
+	for _, objName := range objectsName {
+		input.Objects = append(input.Objects, obs.ObjectToDelete{
+			Key: objName,
+		})
+	}
+	output, err := client.DeleteObjects(input)
 	if err == nil {
-		return nil
+		return output, nil
 	}
 	if obsError, ok := err.(obs.ObsError); ok && obsError.StatusCode == http.StatusNotFound {
-		return status.Errorf(codes.NotFound, "Error, the OBS instance %s does not exist: %v", bucketName, err)
+		return nil, status.Errorf(codes.NotFound, "Error, the OBS instance %s does not exist: %v", bucketName, err)
 	}
-	return status.Errorf(codes.Internal, "Error getting OBS instance %s upload list: %v", bucketName, err)
+	return nil, status.Errorf(codes.Internal, "Error getting OBS instance %s upload list: %v", bucketName, err)
```

Comment:
the log information is not accurate, please optimize

Reply:
done, Modify the error handling logic, whether the deletion of the object is successful is reflected in ouput